adapters/repos/db/shard_debug.go

```diff
@@ -43,18 +43,18 @@ func (s *Shard) DebugResetVectorIndex(ctx context.Context, targetVector string)
 		return errors.Wrap(err, "drop vector index")
 	}
 
-	if s.hasTargetVectors() {
-		s.vectorIndexes[targetVector], err = s.initVectorIndex(ctx, targetVector, s.index.vectorIndexUserConfigs[targetVector])
+	if targetVector == "" {
+		s.vectorIndex, err = s.initVectorIndex(ctx, targetVector, s.index.vectorIndexUserConfig)
```

Comment:
This method is very weird as it mutates the vector index without grabbing any lock. However, this looks as a never used method so did not want to introduce locking with this PR